packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1434,7 +1434,7 @@ class _TabBarViewState extends State<TabBarView> {
     super.didChangeDependencies();
     _updateTabController();
     _currentIndex = _controller?.index;
-    _pageController = PageController(initialPage: _currentIndex ?? 0);
+    _pageController = PageController(initialPage: _currentIndex ?? 0, keepPage: false);
```

Comment:
A comment explaining why `keepPage` is false here will be useful to the future reader.